graphql-language/src/ast.rs

```diff
@@ -7,22 +7,23 @@ use source::Source;
 
 /// Contains some character offsets that identify where the source of the AST
 /// is from.
+#[derive(Clone, Copy)]
 pub struct Location<'a> {
   pub start: usize,
   pub end: usize,
-  pub source: Option<&'a Source<'a>>
+  pub source: Source<'a>
```

Comment:
Source should be optional and *must* be a reference. Since pretty much every node will have a location (and thus a source) we can't make a new copy for every AST node.